.github/workflows/ci.yml

```diff
@@ -0,0 +1,47 @@
+name: test
+
+on:
+  push:
+    branches:
+    - '*'
+  pull_request:
+    branches:
+    - '*'
+
+jobs:
+  prepare:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - id: make-list
+        run: |
+          echo "::set-output name=dirs::$(ls -d */ | jq -R -s -c 'split("\n")[:-1]')"
+    outputs:
+      dirs: ${{steps.make-list.outputs.dirs}}
+  test:
+    needs: prepare
+    strategy:
+      fail-fast: false
+      matrix:
+        example_to_test: ${{fromJson(needs.prepare.outputs.dirs)}}
+        go_version: [1.17.x]
+        kubernetes_version: [1.17.17]
+        # kubernetes_version: [1.17.17, 1.18.20, 1.19.16, 1.20.15, 1.21.14, 1.22.15, 1.23.12, 1.24.6, 1.25.2]
```

Comment:
There is a limitation from GA
> Job matrix - A job matrix can generate a maximum of 256 jobs per workflow run. This limit applies to both GitHub-hosted and self-hosted runners.

Reply:
Wow, I didn't know that :) That's a pity!

Reply:
I think once we fix the existing codebase, we can test all-at-once and use matrix to combine library version with kubernetes version. It will be 3 * 9 = 27 parallel jobs. Which is very much under the limit.